packages/block-library/src/button/edit.js

```diff
@@ -199,6 +200,7 @@ function ButtonEdit( props ) {
 
 	const borderProps = useBorderProps( attributes );
 	const colorProps = useColorProps( attributes );
+	const spacingProps = useSpacingProps( attributes );
```

Comment:
Can you clarify why this is needed here? Ideally padding support is enabled for blocks by just adding a block support flag without touching the block itself at all.

Reply:
I guess this is because of the extra wrapper of the button block? That wrapper is hurting us a lot, ideally we should find ways to remove it without breaking changes. We tried this in the past but reverted. I wonder if we could make it opt-in somehow.

It seems for now we're just accepting the less than ideal button markup.

 cc @mtias

Reply:
Definitely worth considering and exploring. I'm not too sure of what was the case with the extra wrapper.